changelogs/fragments/1927-removed-parameter-invalid.yml

```diff
@@ -0,0 +1,12 @@
+deprecated_features:
+  - composer - deprecated invalid parameter aliases ``working-dir``, ``global-command``, ``prefer-source``, ``prefer-dist``, ``no-dev``, ``no-scripts``, ``no-plugins``, ``optimize-autoloader``, ``classmap-authoritative``, ``apcu-autoloader``, ``ignore-platform-reqs``, will be removed in 5.0.0 (https://github.com/ansible-collections/community.general/pull/1927).
+  - apt_rpm - deprecated invalid parameter alias ``update-cache``, will be removed in 5.0.0 (https://github.com/ansible-collections/community.general/pull/1927).
+  - homebrew - deprecated invalid parameter alias ``update-brew``, will be removed in 5.0.0 (https://github.com/ansible-collections/community.general/pull/1927).
+  - homebrew_cask - deprecated invalid parameter alias ``update-brew``, will be removed in 5.0.0 (https://github.com/ansible-collections/community.general/pull/1927).
+  - opkg - deprecated invalid parameter alias ``update-cache``, will be removed in 5.0.0 (https://github.com/ansible-collections/community.general/pull/1927).
+  - pacman - deprecated invalid parameter alias ``update-cache``, will be removed in 5.0.0 (https://github.com/ansible-collections/community.general/pull/1927).
+  - slackpkg - deprecated invalid parameter alias ``update-cache``, will be removed in 5.0.0 (https://github.com/ansible-collections/community.general/pull/1927).
+  - urmpi - deprecated invalid parameter aliases ``update-cache`` and ``no-recommends``, will be removed in 5.0.0 (https://github.com/ansible-collections/community.general/pull/1927).
+  - xbps - deprecated invalid parameter alias ``update-cache``, will be removed in 5.0.0 (https://github.com/ansible-collections/community.general/pull/1927).
+  - github_deploy_key - deprecated invalid parameter alias ``2fa_token``, will be removed in 5.0.0 (https://github.com/ansible-collections/community.general/pull/1927).
+  - "puppet - deprecated undocumented parameter ``show_diff``, will be removed in 5.0.0. Use ``--diff`` or the task attribute ``diff: true`` instead (https://github.com/ansible-collections/community.general/pull/1927)."
```

Comment:
Since you also removed `module._diff`, this is on longer correct now.

Reply:
You lost me somewhere: I did not remove it. Reading about this, I think I see where you coming from with the `module._diff`, but I believe this module never handled diff-mode properly, to begin with. So I do see that as a worthy improvement, just not in the scope of this one PR.

That said, yes, I am removing the mentions to `--diff` or `diff: yes` from that changelog.

Reply:
Ok, in that case, please increase the deprecation period to say 7.0.0 since we are now removing a (undocumented) feature with no replacement. That gives people more time to notice they actually rely on this feature and complain on its deprecation.

Reply:
done